execution_engine/src/core/runtime_context/mod.rs

```diff
@@ -162,6 +163,53 @@ where
         }
     }
 
+    #[allow(clippy::too_many_arguments)]
+    pub fn fork(
```

Comment:
I'm not sure how I feel about this method.  

I think the old approach of using `RuntimeContext::state` and `RuntimeContext::new` was more direct and a little easier to understand at the use-site.  Now it requires me to understand the semantics of `RuntimeContext::fork` which aren't evident just from the use-site.

Reply:
Removed - initially I was trying to completely avoid accessing RuntimeContext's tracking copy (through removing `.state()` method) but after some refactoring write/add calls into metered call we can reexpose state() safely